.github/workflows/backstage-release-snapshot.yaml

```diff
@@ -0,0 +1,84 @@
+name: Publish snapshot on NPM
+
+on:
+  # When manually triggered
+  workflow_dispatch:
+
+  # When there's a new commit in main branch
+  push:
+    branches: [ "main" ]
+
+jobs:
+  publish-release-snapshot-on-npm:
+    name: Publish release on NPM
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup release name
+        run: |
+          # get the short git hash
+          git_hash=$(git rev-parse --short "$GITHUB_SHA")
+          # get timestamp
+          now=$(date +'%Y%m%d%H%M%S')
+          # build the release name
+          new_version="0.0.0-snapshot.${git_hash}.${now}"
```

Comment:
we were using `0.0.0-nightly` before, now it's snapshot ?

https://www.npmjs.com/package/@knative-extensions/plugin-knative-event-mesh-backend/v/0.0.0-nightly?activeTab=code

Reply:
Snapshot is a better term :) I don't want to do nightly releases. Instead, publishing a snapshot when there's a new commit makes more sense.

And that `nightly` release was never supposed to be published in the first place. I will delete it (needs some waiting time I think).

Reply:
but we use nightly everywhere else, another term will be inconsistent with the backend

Reply:
If the Go side is also doing that way, we should fix it in the knative/hack repo as they're not really nightlies.

Reply:
Now I see what you mean